Library/Homebrew/cmd/tap.rb

```diff
@@ -37,13 +37,17 @@ def tap_args
       switch "--force-auto-update",
              description: "Auto-update tap even if it is not hosted on GitHub. By default, only taps "\
                           "hosted on GitHub are auto-updated (for performance reasons)."
+      switch "--no-force-auto-update",
+             description: "Disable auto-update for a tap."
```

Comment:
```suggestion
      switch "--[no-]force-auto-update",
             description: "Auto-update tap even if it is not hosted on GitHub. By default, only taps "\
                          "hosted on GitHub are auto-updated (for performance reasons)."
```

These can be combined as `--[no-]force-auto-update`.

`args.force_auto_update?` would then either be true (`--force-auto-update`), false (`--no-force-auto-update`) or nil (neither passed), which sounds like exactly what you want and can simplify the logic down below.

Reply:
Thanks!